src/containers/layout/chat-menu/ChatMenu.tsx

```diff
@@ -0,0 +1,153 @@
+  currentChat: ChatResponse
+  onClose: () => void
+  updateChats: () => Promise<void>
+}
+
+const ChatMenu: FC<ChatMenuProps> = ({
+  anchorEl,
+  currentChat,
+  onClose,
+  updateChats
+}) => {
+  const { t } = useTranslation()
+  const { openDialog } = useConfirm()
+  const { setAlert } = useSnackBarContext()
+
+  const onResponse = useCallback(() => {
+    setAlert({
+      severity: snackbarVariants.success,
+      message: 'chatPage.deleteSuccess'
+    })
+  }, [setAlert])
+
+  const onResponseError = useCallback(
+    (error: ErrorResponse) => {
+      setAlert({
+        severity: snackbarVariants.error,
+        message: error ? `errors.${error.code}` : ''
+      })
+    },
+    [setAlert]
+  )
+
+  const markAsDeletedService = useCallback(
+    (id?: string): Promise<AxiosResponse> =>
+      chatService.markChatAsDeleted(id ?? ''),
+    []
+  )
+
+  const deleteChatService = useCallback(
+    (id?: string): Promise<AxiosResponse> => chatService.deleteChat(id ?? ''),
+    []
+  )
+
+  const { fetchData: markAsDeleted } = useAxios({
+    service: markAsDeletedService,
+    defaultResponse: defaultResponses.object,
+    onResponse,
+    onResponseError,
+    fetchOnMount: false
+  })
+
+  const { fetchData: deleteChat } = useAxios({
+    service: deleteChatService,
+    defaultResponse: null,
+    onResponse,
+    onResponseError,
+    fetchOnMount: false
+  })
+
+  const handleDeletion = async (
+    id: string,
+    isConfirmed: boolean,
+    deletingFully: boolean
+  ) => {
+    if (isConfirmed) {
+      if (deletingFully) await deleteChat(id)
+      else await markAsDeleted(id)
+
+      await updateChats()
+    }
+  }
+
+  const onClearHistory = (e: MouseEvent<HTMLButtonElement>) => {
+    e.stopPropagation()
+    onClose()
+  }
+
+  const onDelete = (e: MouseEvent<HTMLButtonElement>, id: string) => {
+    e.stopPropagation()
+    onClose()
+    const deletingFully = currentChat.deletedFor.length > 0
+
+    openDialog({
+      message: deletingFully
+        ? 'chatPage.fullDeleteWarning'
+        : 'chatPage.markingAsDeletedWarning',
+      sendConfirm: (isConfirmed: boolean) =>
+        void handleDeletion(id, isConfirmed, deletingFully),
+      title: deletingFully
+        ? 'chatPage.fullDeleteTitle'
+        : 'chatPage.markingAsDeletedTitle'
+    })
+  }
+
+  const menuButtons = [
+    {
+      _id: 1,
+      icon: <UpdateDisabledIcon />,
+      name: t('chatPage.clearHistory'),
+      handleOnClick: (e: MouseEvent<HTMLButtonElement>) => onClearHistory(e)
+    },
+    {
+      _id: 2,
+      icon: <DeleteOutlineIcon />,
+      name: t('chatPage.deleteChat'),
+      handleOnClick: (e: MouseEvent<HTMLButtonElement>) =>
+        onDelete(e, currentChat._id),
+      isDangerous: true
+    }
+  ]
+
+  const menuItems = menuButtons.map((item) => (
+    <MenuItem
+      component={ComponentEnum.Button}
+      key={item._id}
+      onClick={item.handleOnClick}
+      sx={styles.menuItem(item.isDangerous ?? false)}
```

Comment:
I guess ?? false is redundant here, if isDangerous is undefined it will be false

Reply:
> I guess ?? false is redundant here, if isDangerous is undefined it will be false

yup, I wanted to remove it too, but typescript is going mad because `sx` doesn't want to receive undefined

Reply:
use !! to convert it to boolean or Boolean(Item.isDangerous)

Reply:
> use !! to convert it to boolean or Boolean(Item.isDangerous)
thanks, fixed!